src/backend/replication/walsender.c

```diff
@@ -2169,36 +2191,60 @@ ProcessStandbyHSFeedbackMessage(void)
  * If wal_sender_timeout is enabled we want to wake up in time to send
  * keepalives and to abort the connection if wal_sender_timeout has been
  * reached.
+ *
+ * If wal archiving and streaming of last archived wal to standby is enabled
+ * we want to wake up in time to send archival report.
  */
 static long
 WalSndComputeSleeptime(TimestampTz now)
 {
-	long		sleeptime = 10000;	/* 10 s */
+	long		sleeptime;
+	TimestampTz wakeup_time;
+	TimestampTz w;
 
-	if (wal_sender_timeout > 0 && last_reply_timestamp > 0)
-	{
-		TimestampTz wakeup_time;
+	wakeup_time = TimestampTzPlusMilliseconds(now, 10000);
 
+	if (wal_sender_timeout > 0 && last_reply_timestamp > 0) {
 		/*
 		 * At the latest stop sleeping once wal_sender_timeout has been
 		 * reached.
-		 */
-		wakeup_time = TimestampTzPlusMilliseconds(last_reply_timestamp,
-												  wal_sender_timeout);
-
-		/*
+		 *
 		 * If no ping has been sent yet, wakeup when it's time to do so.
 		 * WalSndKeepaliveIfNecessary() wants to send a keepalive once half of
 		 * the timeout passed without a response.
 		 */
-		if (!waiting_for_ping_response)
-			wakeup_time = TimestampTzPlusMilliseconds(last_reply_timestamp,
-													  wal_sender_timeout / 2);
+		if (waiting_for_ping_response)
+			w = TimestampTzPlusMilliseconds(last_reply_timestamp,
+											wal_sender_timeout);
+		else
+			w = TimestampTzPlusMilliseconds(last_reply_timestamp,
+											wal_sender_timeout / 2);
 
-		/* Compute relative time until wakeup. */
-		sleeptime = TimestampDifferenceMilliseconds(now, wakeup_time);
+		if (w < wakeup_time)
+			wakeup_time = w;
 	}
 
+	if (XLogArchivingActive() && wal_sender_archiving_status_interval > 0)
+	{
+		/*
+		* If we requested an update from pgstat, poll every
+		* ARCHIVE_REQUEST_INTERVAL for the result. Otherwise wait until it's
+		* time to send a new report.
+		*/
```

Comment:
This needs to be formatted correctly.
```suggestion
		/*
		 * If we requested an update from pgstat, poll every
		 * ARCHIVE_REQUEST_INTERVAL for the result. Otherwise wait until it's
		 * time to send a new report.
		 */
```